frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetailsColumns.tsx

```diff
@@ -0,0 +1,219 @@
+/* Copyright Contributors to the Open Cluster Management project */
+import { Badge, LabelGroup, Skeleton } from '@patternfly/react-core'
+import { ListItems } from '../../../../ui-components'
+import { generatePath, Link } from 'react-router-dom-v5-compat'
+import { NavigationPath } from '../../../../NavigationPath'
+import { ExternalLinkAltIcon } from '@patternfly/react-icons'
+import { TFunction } from 'react-i18next'
+import { ReactNode } from 'react'
+import { collectKinds } from '../../common/util'
+import { Grid as MuiGrid } from '@mui/material'
+
+interface IKinds {
+  apiGroups: string[]
+  kinds: string[]
+}
+// Any resources that are related to Vapb then add these rows
+export const addRowsForHasVapb = (
+  cols: ListItems[],
+  hasVapb: boolean,
+  loading: boolean,
+  vapbItems: any[] | null | undefined,
+  apiGroup: string,
+  clusterName: string,
+  name: string
+): ListItems[] => {
+  if (hasVapb) {
+    // Loading to fetch VAPB
+    if (!vapbItems) {
```

Comment:
Should this also check if it's loading?

Reply:
Maybe checking if it's null or undefined would be better.

Reply:
Which variable?

Reply:
I think `if (!vapbItems) {` would be true if it's an empty array so it'd be nice to guard against that.

Reply:
I was wrong. I thought an empty array was falsy but apparently not:
```
> if(![]){console.log("hello")}
undefined
> if([]){console.log("hello")}
hello
```

Reply:
vapbItems = [] then vapbItems is considered true in if condition

Reply:
let me change to loading

Reply:
But there would be little blicking because of loading (init is false so)